ROX-24744: Update docs with namespace exclusions for admission controller (#100954)
openshift/openshift-docs: modules/namespace-exclusions.adoc

:_mod-docs-content-type: CONCEPT
[id="namespace-exclusions_{context}"]
= Namespace exclusions from admission controller enforcement

[role="_abstract"]

> By default, {product-title-short} excludes certain administrative namespaces, such as the `stackrox`, `kube-system`, and `istio-system` namespaces, from admission controller enforcement blocking. The reason for this is that some items in these namespaces must be deployed for {product-title-short} to work correctly. 

Reviewer: ```suggestion
By default, {rh-rhacs-first} excludes the following administrative namespaces from admission controller enforcement blocking: 
`stackrox`
`kube-system` 
`istio-system`
This exclusion is necessary because these namespaces contain some items that you must deploy for {product-title-short} to work correctly.
```

> These exclusions mean that API calls for these namespaces are not passed to {product-title-short} for review. Instead of rejecting the `CREATE` verb, the deployment is created and then scaled down using Sensor. 

Reviewer: ```suggestion
These exclusions mean that {product-title-short} does not review API calls for these namespaces. Instead of rejecting the `CREATE` verb, Sensor creates the deployment and then scales it down.
```

Author: Sensor does not create the deployment - rewritten

:_mod-docs-content-type: CONCEPT
[id="namespace-exclusions_{context}"]
= Namespace exclusions from admission controller enforcement

[role="_abstract"]
By default, {product-title-short} excludes certain administrative namespaces, such as the `stackrox`, `kube-system`, and `istio-system` namespaces, from admission controller enforcement blocking. The reason for this is that some items in these namespaces must be deployed for {product-title-short} to work correctly. 

These exclusions mean that API calls for these namespaces are not passed to {product-title-short} for review. Instead of rejecting the `CREATE` verb, the deployment is created and then scaled down using Sensor. 

There are two places where excluded namespaces are defined:

* The configuration of the `ValidatingWebhookConfiguration`
* The {product-title-short} code that defines "system" namespaces

The following namespaces are excluded:

> * `stackrox`

Author: and the second place he links to is in the code: https://github.com/stackrox/stackrox/blob/master/pkg/kubernetes/namespaces.go

> * `stackrox`

Reviewer 2: In that case the documented list is incorrect. The correct list is:
* `kube-system`
* `kube-public`
* `istio-system`
* `stackrox`

:_mod-docs-content-type: CONCEPT
[id="namespace-exclusions_{context}"]
= Namespace exclusions from admission controller enforcement

[role="_abstract"]

> By default, {rh-rhacs-first} excludes certain administrative namespaces from admission controller enforcement blocking. The reason for this is that some items in these namespaces must be deployed for {product-title-short} to work correctly. 

Reviewer 2: ... from the validating webhook configurations in the admission controller. Policy evaluation and enforcement is not performed on review requests originating from these administrative namespaces.

> These exclusions mean that API calls for these namespaces are not passed to the admission controller for review. Instead of rejecting the `CREATE` verb, {product-title-short} allows the deployment to be created, but then it is scaled down using Sensor. 

Reviewer 2: Suggested removal: Beginning from Instead of ....

Also given my comment above, the line "These exclusions ..." could also be deleted
